File: studentgradechecker-classifygrade/src/main/java/com/webservice/GradeCalculator.java
package com.webservice;

import java.util.ArrayList;

public class GradeCalculator {

    private static final double SEVENTY = 70.0;
    private static final double SIXTY = 60.0;
    private static final double FIFTY = 50.0;
    private static final double FORTY = 40.0;
    private static final double THIRTY = 30.0;

    private static final String DISTINCTION = "Distinction";
    private static final String COMMENDATION = "Commendation";
    private static final String PASS = "Pass";
    private static final String MARGINAL_FAIL = "Marginal Fail";
    private static final String FAIL = "Fail";
    private static final String LOW_FAIL = "Low Fail";
    private static final String NEWLINE_REGEX = "newline";
    private static final String EMPTY_STRING = "";
    private static final String COMMA_REGEX = ".*,";
    private static final String FORWARD_SLASH_REGEX = "/";

    /**
     * Private constructor for utility Controller class to prevent creation of object.
     */
    private GradeCalculator(){}

   /**
     * Parse marks entered by student from input String.
     * 
     * @param inputText String input of modules and marks from request.
     * @return ArrayList<String> of marks without module names or other characters
     */
    public static ArrayList<String> parseMarksFromInput(String inputText){
      
      String cleanInput = inputText.replace(NEWLINE_REGEX, FORWARD_SLASH_REGEX);
   
      String [] lines = cleanInput.split(FORWARD_SLASH_REGEX);
   
      ArrayList<String> marks = new ArrayList<>();
   
      for(String line : lines){
        marks.add(line.replaceAll(COMMA_REGEX, EMPTY_STRING));
      }
   
      return marks;
    }
   
    /**
     * Calculate final mark from input marks provided.
     * 
     * Final mark is calculated by dividing the total sum of marks 
     * by the number of modules.
     * 
     * @param marks ArrayList<String> of marks used to calculate total.
     * @return double of calculated final mark.
     */
    public static double calculateFinalMarkFromMarks(ArrayList<String> marks){
      int total = 0;
      double finalMark = 0;
      
      for(String mark: marks){
        total += Integer.parseInt(mark);
      }
   
      finalMark = total / marks.size();
   
      return finalMark;
    }
   
    /**
     * Classify student grade based on final mark.
     * 
     * @param finalMark double final mark score for all modules.
     * @return String of classified grade.
     */
    public static String classifyGrade(double finalMark){
      String grade = EMPTY_STRING;
   
      if(finalMark >= SEVENTY){
        grade = DISTINCTION;
      } else if (finalMark >= SIXTY && finalMark < SEVENTY) {
          grade = COMMENDATION;
      } else if (finalMark >= FIFTY && finalMark < SIXTY) {
          grade = PASS;
      } else if (finalMark >= FORTY && finalMark < FIFTY) {
          grade = MARGINAL_FAIL;
      } else if (finalMark >= THIRTY && finalMark < FORTY) {
          grade = FAIL;
      } else {
          grade = LOW_FAIL;
      }
   
      return grade;
   
    }
    
}

File: studentgradechecker-classifyeach/src/main/kotlin/com/webservice/GradeCalculator.kt
package com.webservice

import java.util.*

object GradeCalculator {
    private const val SEVENTY = 70.0
    private const val SIXTY = 60.0
    private const val FIFTY = 50.0
    private const val FORTY = 40.0
    private const val THIRTY = 30.0
    private const val DISTINCTION = "Distinction"
    private const val COMMENDATION = "Commendation"
    private const val PASS = "Pass"
    private const val MARGINAL_FAIL = "Marginal Fail"
    private const val FAIL = "Fail"
    private const val LOW_FAIL = "Low Fail"
    private const val NEWLINE_REGEX = "newline"
    private const val EMPTY_STRING = ""
    private const val COMMA_REGEX = ".*,"
    private const val FORWARD_SLASH_REGEX = "/"

    /**
     * Parse marks entered by student from input String.
     *
     * @param inputText String input of modules and marks from request.
     * @return ArrayList<String> of marks without module names or other characters
    </String> */
    @JvmStatic
    fun parseMarksFromInput(inputText: String): MutableList<Double> {
        val cleanInput = inputText.replace(NEWLINE_REGEX, FORWARD_SLASH_REGEX)
        val lines = cleanInput.split(FORWARD_SLASH_REGEX.toRegex()).toTypedArray()

        val marks: MutableList<Double> = ArrayList()

        for (line in lines) {
            val mark = line.replace(COMMA_REGEX.toRegex(), EMPTY_STRING)
            marks.add(mark.toDouble())
        }
        return marks
    }

    /**
     * Classify student grade based on final mark.
     *
     * @param finalMark double final mark score for all modules.
     * @return String of classified grade.
     */
    @JvmStatic
    fun classifyGrades(finalMark: Double): String {

        var grade: String = if (finalMark >= SEVENTY) {
            DISTINCTION
        } else if (finalMark >= SIXTY && finalMark < SEVENTY) {
            COMMENDATION
        } else if (finalMark >= FIFTY && finalMark < SIXTY) {
            PASS
        } else if (finalMark >= FORTY && finalMark < FIFTY) {
            MARGINAL_FAIL
        } else if (finalMark >= THIRTY && finalMark < FORTY) {
            FAIL
        } else {
            LOW_FAIL
        }
        return grade
    }
}